### PR TITLE
Update Jenkins plugin docs: add agent ports connectivity info; add diagnostic flare instructions

### PR DESCRIPTION
**Pull request by nikita-tkachenko-datadog**
### What does this PR do? What is the motivation?

Updates Datadog Jenkins plugin documentation:
- Agent installation instructions now include the list of ports that the plugin needs to have access to
- Troubleshooting instructions now include the steps to generate plugin diagnostic flare
- Some minor corrections around the agentless/agentful installations are done

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 1]
```diff
@@ -1513,3 +1525,5 @@ Failed to reinitialize Datadog-Plugin Tracer, Cannot enable traces collection vi
 [27]: /logs/guide/best-practices-for-log-management/
 [28]: /continuous_integration/search/#search-for-pipelines
 [29]: /agent/logs/?tab=tcpudp#custom-log-collection
+[30]: /developers/dogstatsd/
+[31]: https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml
```
Comment (hestonhoffman) [inline content/en/continuous_integration/pipelines/jenkins.md:1529]:
```suggestion
[31]: /containers/docker/apm/#tracing-from-the-host
```
I think this config template link might be a little confusing for users. How do you feel about this link instead?

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 2]
```diff
@@ -1407,6 +1417,16 @@ The **CI Pipeline List** page shows data for only the default branch of each rep
 
 ## Troubleshooting
 
+### Generate diagnostic flare
+
+When reporting an issue to the Datadog support team, please generate a plugin diagnostic flare and provide it along with the issue description.
```
Comment (hestonhoffman) [inline content/en/continuous_integration/pipelines/jenkins.md:1422]:
```suggestion
When reporting an issue to the Datadog support team, generate a plugin diagnostic flare and provide it along with the issue description.
```

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 3]
```diff
@@ -1407,6 +1417,16 @@ The **CI Pipeline List** page shows data for only the default branch of each rep
...
+
+To generate the flare do the following:
+
+1. In your Jenkins instance web interface, go to **Manage Jenkins > Troubleshooting > Datadog**.
+2. In the Diagnostic Flare form check which information you want to include in the flare (default selection works best, the more information provided the easier it is to diagnose your issue).
```
Comment (hestonhoffman) [inline content/en/continuous_integration/pipelines/jenkins.md:1427]:
```suggestion
2. In the Diagnostic Flare form, check which information you want to include in the flare. The default selection works best. The more information you provide, the easier it is to diagnose your issue.
```

File: content/en/continuous_integration/pipelines/jenkins.md [hunk 4]
```diff
@@ -1407,6 +1417,16 @@ The **CI Pipeline List** page shows data for only the default branch of each rep
...
+3. Click `Download` to generate and download the flare archive.
```
Comment (hestonhoffman) [inline content/en/continuous_integration/pipelines/jenkins.md:1428]:
```suggestion
3. Click **Download** to generate and download the flare archive.
```

**Review by hestonhoffman — COMMENTED** [review]
Thanks for the docs! Left some suggestions

File: content/en/continuous_integration/pipelines/jenkins.md (hunk 1 above)
Comment (nikita-tkachenko-datadog) [reply content/en/continuous_integration/pipelines/jenkins.md:1529]: Agree about the template being confusing. I couldn't find a suitable link, but this one seems like a good fit. Thanks!

**Review by hestonhoffman — APPROVED** [review]
LGTM!

**Comment by hestonhoffman** [comment]
@nikita-tkachenko-datadog Asked for a review in #software-delivery for the required reviews above.

**Comment by nikita-tkachenko-datadog** [comment]
@hestonhoffman this is ready for merge now, could you please merge it? Thanks!

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**